src/metrics/histogram.rs

```diff
@@ -122,6 +122,27 @@ pub fn exponential_buckets(start: f64, factor: f64, length: u16) -> impl Iterato
         .take(length.into())
 }
 
+/// Exponential bucket distribution within a range
+/// /// Creates `length` buckets, where the lowest bucket is `min` and the highest bucket is `max`.
+/// /// The final +Inf bucket is not counted and not included in the returned iterator.
+/// /// The function panics if `length` is 0 or negative, or if `min` is 0 or negative.
+fn exponential_buckets_range(min: f64, max: f64, length: u16) -> impl Iterator<Item = f64> {
+    if length < 1 {
+        panic!("ExponentialBucketsRange length needs a positive length");
+    }
```

Comment:
You could use a `NonZeroU16` instead of a `u16`. That said, that would be inconsistent with `exponential_buckets`. Thoughts?

Reply:
I guess better to keep as u16 to keep consistent with `exponential_buckets`.
We can keep the additional check in the code for correctness.
What do you think?